searx/engines/public_domain_image_archive.py

```diff
@@ -59,61 +57,52 @@ def _clean_url(url):
     return urlunparse((parsed.scheme, parsed.netloc, parsed.path, parsed.params, urlencode(query), parsed.fragment))
 
 
-def _get_algolia_api_key():
-    global __CACHED_API_KEY  # pylint:disable=global-statement
+def _get_algolia_api_url():
+    global __CACHED_API_URL  # pylint:disable=global-statement
 
-    if __CACHED_API_KEY:
-        return __CACHED_API_KEY
+    if __CACHED_API_URL:
+        return __CACHED_API_URL
 
-    resp = get(pdia_search_url)
+    # fake request to extract api url
+    resp = get(f"{pdia_base_url}/search/?q=")
     if resp.status_code != 200:
-        raise LookupError("Failed to fetch config location (and as such the API key) for PDImageArchive")
+        raise LookupError("Failed to fetch config location (and as such the API url) for PDImageArchive")
     pdia_config_filepart = extr(resp.text, pdia_config_start, pdia_config_end)
     pdia_config_url = pdia_base_url + pdia_config_start + pdia_config_filepart + pdia_config_end
 
     resp = get(pdia_config_url)
     if resp.status_code != 200:
-        raise LookupError("Failed to obtain Algolia API key for PDImageArchive")
+        raise LookupError("Failed to obtain AWS api url for PDImageArchive")
 
-    api_key = extr(resp.text, 'const r="', '"', default=None)
+    api_url = extr(resp.text, 'const r="', '"', default=None)
 
-    if api_key is None:
-        raise LookupError("Couldn't obtain Algolia API key for PDImageArchive")
+    if api_url is None:
+        raise LookupError("Couldn't obtain AWS api url for PDImageArchive")
 
-    __CACHED_API_KEY = api_key
-    return api_key
+    __CACHED_API_URL = api_url
+    return api_url
 
 
-def _clear_cached_api_key():
-    global __CACHED_API_KEY  # pylint:disable=global-statement
+def _clear_cached_api_url():
+    global __CACHED_API_URL  # pylint:disable=global-statement
 
-    __CACHED_API_KEY = None
+    __CACHED_API_URL = None
 
 
 def request(query, params):
-    api_key = _get_algolia_api_key()
-
-    args = {
-        'x-algolia-api-key': api_key,
-        'x-algolia-application-id': 'OQI2J6V4IZ',
-    }
-    params['url'] = f"{base_url}/1/indexes/*/queries?{urlencode(args)}"
-    params["method"] = "POST"
-
-    request_params = {
-        "page": params["pageno"] - 1,
-        "query": query,
-        "highlightPostTag": "__ais-highlight__",
-        "highlightPreTag": "__ais-highlight__",
-    }
-    data = {
-        "requests": [
-            {"indexName": "prod_all-images", "params": urlencode(request_params)},
-        ]
+    params['url'] = _get_algolia_api_url()
+    params['method'] = 'POST'
+
+    request_data = {
+        'page': params['pageno'] - 1,
+        'query': query,
+        'hitsPerPage': page_size,
+        'indexName': 'prod_all-images',
     }
-    params["data"] = dumps(data)
+    params['headers'] = {'Content-Type': 'application/json', 'User-Agent': gen_useragent()}
```

Comment:
The UA is already set in the online processor ..

https://github.com/searxng/searxng/blob/f798ddd4922d793d5e6ccb7c4111810d549ff4f4/searx/search/processors/online.py#L62-L63

and the `params['headers']` is already a dict we should not overwrite here / set your content type by:

```python
param[header]['Content-Type'] = "application/json"
```
I

Reply:
Good catch, actually many other enginesuse `gen_useragent` too although it's redundant. Might be worth a new issue.